### ISSUE TITLE
[Suggestion] Arcade / Novice mode & HUD

### ISSUE THREAD

**Issue by mgavioli** [leezer3/OpenBVE#37]
I expect this suggestion to be frowned upon by the expert drivers! But nobody is born expert, and even the expert might stumble upon the route he is unfamiliar with...

While learning a yet unknown (to me) route, I often would like a kind of HUD with a number of info, clearly displayed and easy to spot, possibly not too far away from the track view. For instance:
- current speed (already there, but hard to read while keeping an eye on the track)
- current clock (same as above)
- current track / signal speed limit
- distance to next speed limit post (and maybe the limit itself)
- distance to next station (with name? with ETA?)
- distance to next non-green signal?

I understand that this would be a sort of 'gamification' of the simulator, but simulators running on PC lack a great part of the 'look and feel' of real-life operations (having spent twenty years in flight simulation, I am rather confident about this) and some 'bridge' might be useful while learning and/or could attract more users. Of course, nobody would be forced to use it!

At the same time, I propose to rename the "Arcade mode" into "Novice mode" (or whatever other English term seems suited to native English speakers), in which this HUD would be the default.

Comments? Indecent? Way off? Tolerable? Useful?

**Comment by JakubVanek** [comment]
> - track / signal speed limit
> - distance to next speed limit post (and maybe the limit itself)
YES! Sometimes I have problem to see where the speed limit starts to apply. AI "magically" knows the exact place, so I make use of that.

**Comment by mgavioli** [comment]
How do you see such a feature could be implemented?
- as an extension of the current HUD system (`CurrentHudElements` array as managed in `Overlays.cs`)
- as a new independent mode, in addition to normal, debug, menu, ... modes?
- should it be possible to turn on/off individual elements (for instance, "show me the current speed limit, but don't show me next station info")? Or it should be 'all or nothing'? My impression is that the former would be confusing, but it may appeal to someone.

**Comment by leezer3** [comment]
I'd be tempted to implement as a third independent HUD mode, on the basis that it should replace the current HUD messages in this mode.

Turning on and off is a little more problematic-
Providing a 'fancy' GUI is work for the translators, and for whoever programs the UI itself.
Possibly the best way to handle this is probably to add a new section to the Options.cfg , and allow turning options off in there, but by default have everything on.
You could then use the same basic principle as the CurrentHUDElements- Render the elements in the array in descending order.

**Comment by Hibiii** [comment]
I agree with renaming arcade mode to novice mode. I'm not knowledgeable on HUDs, but I'd advise to expose more subjects to the HUD interface as a start. Then, it would be merely a question of adding elements to existing HUDs! Expert mode doesn't show the stop indicators, normal mode deosn't show the score. It could me similarly used for the suggested elements, yo.
